### ISSUE TITLE
Enhancement : Content spanning on multiple slides

### ISSUE THREAD

**Issue by GrmpCerber** [asciidoctor/asciidoctor-reveal.js#68]
I've got a document which content is suitable for both PDF and reveal presentation.
Yet, as pointed out by #59 title level 3 and below are ignored, thus ending in content being unreadable.

In addition to #59 I would say that "level" based splitting is not enough: as an example, I’ve got a section that holds a list and two schemas. 
Only the list part is visible. 
I would like to be able to show the two schema on different slides without altering the PDF display (thus not using `<<<` as slide splitter)

**Comment by obilodeau** [comment]
What version of asciidoctor-reveal.js are you using? Current master has this in https://github.com/asciidoctor/asciidoctor-reveal.js/commit/f027d08cd22a5e1ca27a46b96dd380abe062ba0a which means that level 3 and below are no longer ignored and should render in your PDF.

As for your second remark, can you give a short example in AsciiDoc? I don't understand.

**Comment by GrmpCerber** [comment]
I've discovered one could create a fake section by adding 
    
    === !
    
(Preceded and followed by a blank line)
Yet this is not practical as it impact the PDF document ...

@obilodeau I did a shallow clone yesterday so my version is d39f64d + Asciidoctor 1.5.4

As for a short example I did so in the attached file 
[test.adoc.txt](https://github.com/asciidoctor/asciidoctor-reveal.js/files/236804/test.adoc.txt) and compiled it with : 

    asciidoctor -r asciidoctor-diagram -T ./asciidoctor-reveal.js/templates/slim/ test.adoc

My vertical slide visually ends with the level 3 title.
As you could guess my example could be visually splited in 4 slides : 
* level 2 title and explaination 
* level 3 title with list 
* and then the two schemas on two slides

**Comment by mojavelinux** [comment]
> I've discovered one could create a fake section..., Yet this is not practical as it impact the PDF document ...

Keep in mind you can use preprocessor conditionals to hide these fake sections from other backends:

```
ifndef::backend-pdf[=== !]
```

You can also do all sorts of other things using preprocessor conditionals.

**Comment by GrmpCerber** [comment]
Nice !
This should be added to the readme (you or I)

This said, it's not 100% practical since `=== !` are specific to the reaveal.js backend, hence I would suggest adding a `backend-revealjs` attribute.
In addition It would also be a nice feature to be able to repeat the last title (with `=== !repeat` for example)

**Comment by mojavelinux** [comment]
> hence I would suggest adding a backend-revealjs attribute.

At the moment, there are some drawbacks to setting the backend to revealjs. I've explained elsewhere that this would be permissible if the reveal.js converter is setup the same as the Bespoke.js converter. See https://github.com/asciidoctor/asciidoctor-bespoke/blob/master/lib/asciidoctor-bespoke/converter.rb.

In other words, you shouldn't set the backend when using templates only. The backend should only be set if there's a formal converter class to load. Otherwise, it doesn't inherit the correct environment internally.

As a workaround, simply set a custom attribute to use in the preprocessor conditionals, such as `-a revealjs`.

**Comment by mojavelinux** [comment]
> In addition It would also be a nice feature to be able to repeat the last title (with === !repeat for example)

This can be accomplished using document attributes. Set the title to an attribute, then reference the attribute in the title.

**Comment by GrmpCerber** [comment]
I got the idea on how to repeat a title. Not as practical as my suggestion but yours is not so bad and works "out of the box".

Now, concerning the backend definition, I don't fully understand your point. Yet, I'll take your word "as is".
Concerning the attribute name you suggest `revealjs`.
What would occur if I was to choose `-a backend-revealjs` (which IMHO is more explicit)

**Comment by obilodeau** [comment]
> At the moment, there are some drawbacks to setting the backend to revealjs. I've explained elsewhere that this would be permissible if the reveal.js converter is setup the same as the Bespoke.js converter. See https://github.com/asciidoctor/asciidoctor-bespoke/blob/master/lib/asciidoctor-bespoke/converter.rb.
> 
> In other words, you shouldn't set the backend when using templates only. The backend should only be set if there's a formal converter class to load. Otherwise, it doesn't inherit the correct environment internally.

That class holds a lot of code that I don't understand. I'll start something in a branch in this repo and hopefully you'll be able to quickly fix my mistakes.

Meanwhile I'll merge his documentation.

**Comment by obilodeau** [comment]
This was fixed when #80 and #69 were merged. Re-open if I'm mistaken.

**Comment by mojavelinux** [comment]
> That class holds a lot of code that I don't understand.

A lot of it has to do with configuring Slim and allowing the template to be specified per slide. If you look past those lines, what you see remaining is hopefully more straightforward.

```ruby
module Asciidoctor
module Revealjs
class Converter < ::Asciidoctor::Converter::CompositeConverter
  ProvidedTemplatesDir = ::File.expand_path '../../../templates', __FILE__
  register_for 'reveal.js'

  def initialize backend, opts = {}
      # merge user templates with provided templates (user wins)
      template_dirs = [ProvidedTemplatesDir]
      if (user_template_dirs = opts[:template_dirs])
        template_dirs += user_template_dirs.map {|d| ::File.expand_path d }
      end
      # create the main converter
      template_converter = ::Asciidoctor::Converter::TemplateConverter.new backend,
          template_dirs,
          (opts.merge htmlsyntax: 'html', template_engine: 'slim')
      # create the delegate / fallback converter
      html5_converter = ::Asciidoctor::Converter::Html5Converter.new backend, opts
      # fuse the converters together
      super backend, template_converter, html5_converter
      basebackend 'html'
      htmlsyntax 'html'
  end
end
end
end
```

(Obviously, we could improve the API in Asciidoctor here...but this was the first time I used the API).
